internal/service/plan.go

```diff
@@ -423,7 +427,44 @@ func (s *planService) UpdatePlan(ctx context.Context, id string, req dto.UpdateP
 }
 
 func (s *planService) DeletePlan(ctx context.Context, id string) error {
-	err := s.planRepo.Delete(ctx, id)
+
+	entitlements, err := s.entitlementRepo.List(ctx, &types.EntitlementFilter{
+		QueryFilter: types.NewNoLimitQueryFilter(),
+		PlanIDs:     []string{id},
+	})
```

Comment:
also add status published filter here

Reply:
Also - I think we need to check only for subscriptions and not for entitlements.
Entitlements can not be deleted till they are linked to plans
and now if we do not allow deleting plans with entitlements - we end up in a deadlock.